src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -13,7 +28,24 @@ public class UniversityApp {
      * @param lastName  - nazwisko prowadzącego
      */
     public void createLecturer(int id, String degree, String firstName, String lastName) {
+        Lecturer lecturer = findLecturer(id);
+        if (lecturer != null) {
+            System.out.println("Prowadzący z id " + id + " już istnieje");
+        } else {
+            lecturers[lectureCount] = new Lecturer(firstName, lastName, id, degree);
+            lectureCount++;
+        }
+    }
 
+    private Lecturer findLecturer(int id) {
+        for (int i = 0; i < lectureCount; i++) {
+            if (lecturers[i] == null) {
+                return null;
+            } else if (Objects.equals(lecturers[i].getId(), id)) {
+                return lecturers[i];
+            }
+        }
```

Comment:
Tu wystarczy 
```
for (int i = 0; i < lectureCount; i++) {
    if (id == lecturers[i].getId()) {
        return lecturers[i];
    }
}
```
dzięki temu, że nie używasz for-eacha, to nie ma potrzeby zabezpieczania się przed nullem. inty wystarczy porównać przez ==